--- ngx-fancyindex-fd3950172a9e6595ad9ec68c11600e2afe6a2674/ngx_http_fancyindex_module.c	2013-04-22 20:56:57.000000000 +0900
+++ ngx-fancyindex-fd3950172a9e6595ad9ec68c11600e2afe6a2674/ngx_http_fancyindex_module.c	2013-09-26 23:43:21.079507778 +0900
@@ -52,7 +52,6 @@
 
 
 #define NGX_HTTP_FANCYINDEX_PREALLOCATE  50
-#define NGX_HTTP_FANCYINDEX_NAME_LEN     50
 
 
 /**
@@ -563,8 +562,8 @@
         len += ngx_sizeof_ssz("<tr class=\"X\"><td><a href=\"")
             + entry[i].name.len + entry[i].escape /* Escaped URL */
             + ngx_sizeof_ssz("\">")
-            + entry[i].name.len + entry[i].utf_len
-            + NGX_HTTP_FANCYINDEX_NAME_LEN + ngx_sizeof_ssz("&gt;")
+            + entry[i].name.len + 1
+            + ngx_sizeof_ssz("&gt;")
             + ngx_sizeof_ssz("</a></td><td>")
             + 20 /* File size */
             + ngx_sizeof_ssz("</td><td>")
@@ -618,36 +617,14 @@
         *b->last++ = '"';
         *b->last++ = '>';
 
-        len = entry[i].utf_len;
-
-        if (entry[i].name.len - len) {
-            if (len > NGX_HTTP_FANCYINDEX_NAME_LEN) {
-                copy = NGX_HTTP_FANCYINDEX_NAME_LEN - 3 + 1;
-            } else {
-                copy = NGX_HTTP_FANCYINDEX_NAME_LEN + 1;
-            }
-
-            b->last = ngx_utf8_cpystrn(b->last, entry[i].name.data,
-                                          copy, entry[i].name.len);
-            last = b->last;
-
-        } else {
-            b->last = ngx_cpystrn(b->last, entry[i].name.data,
-                                  NGX_HTTP_FANCYINDEX_NAME_LEN + 1);
-            last = b->last - 3;
-        }
-
-        if (len > NGX_HTTP_FANCYINDEX_NAME_LEN) {
-            b->last = ngx_cpymem_ssz(last, "..&gt;</a></td><td>");
-
-        } else {
-            if (entry[i].dir && NGX_HTTP_FANCYINDEX_NAME_LEN - len > 0) {
-                *b->last++ = '/';
-                len++;
-            }
-
-            b->last = ngx_cpymem_ssz(b->last, "</a></td><td>");
+        b->last = ngx_cpystrn(b->last, entry[i].name.data,
+                              entry[i].name.len + 1);
+        last = b->last;
+        if (entry[i].dir) {
+           *b->last++ = '/';
+           len++;
         }
+        b->last = ngx_cpymem_ssz(b->last, "</a></td><td>");
 
         if (alcf->exact_size) {
             if (entry[i].dir) {
