diff -ru a/lib/coroutine.c b/lib/coroutine.c
--- work/accord-9999/lib/coroutine.c    2012-05-20 12:50:49.071294598 +0900
+++ accord-9999/lib/coroutine.c 2012-05-20 12:49:14.116332546 +0900
@@ -20,6 +20,10 @@
  *   Copyright (C) 2011 Kevin Wolf <kwolf@redhat.com>
  */

+/* XXX Is there a nicer way to disable glibc's stack check for longjmp? */
+#ifdef _FORTIFY_SOURCE
+#undef _FORTIFY_SOURCE
+#endif
 #include <stdlib.h>
 #include <setjmp.h>
 #include <stdint.h>
