--- properties/nm-openvpn-editor.c.orig	2017-01-02 10:42:48.074602472 +0900
+++ properties/nm-openvpn-editor.c	2017-01-02 10:42:54.078577593 +0900
@@ -100,7 +100,7 @@
 				goto out;
 		}
 		/* check proto */
-		if (proto && strcmp (proto, "udp") && strcmp (proto, "tcp"))
+		if (proto && strcmp (proto, "udp") && strcmp (proto, "tcp") && strcmp (proto, "udp6"))
 			goto out;
 	}
 	success = TRUE;
--- src/nm-openvpn-service.c.orig	2017-01-02 10:31:10.689715335 +0900
+++ src/nm-openvpn-service.c	2017-01-02 10:31:52.893554394 +0900
@@ -1225,7 +1225,7 @@
 				add_openvpn_arg (args, "1194"); /* default IANA port */
 
 			if (proto) {
-				if (!strcmp (proto, "udp"))
+				if (!strcmp (proto, "udp") || !strcmp (proto, "udp6"))
 					add_openvpn_arg (args, proto);
 				else if (!strcmp (proto, "tcp"))
 					add_openvpn_arg (args, "tcp-client");
