Index: ROCm-OpenCL-Runtime-rocm-5.1.3/CMakeLists.txt
===================================================================
--- ROCm-OpenCL-Runtime-rocm-5.1.3.orig/CMakeLists.txt
+++ ROCm-OpenCL-Runtime-rocm-5.1.3/CMakeLists.txt
@@ -27,7 +27,6 @@
   find_package(OpenCL REQUIRED)
 endif()
 add_subdirectory(amdocl)
-add_subdirectory(tools/clinfo)
 add_subdirectory(tools/cltrace)
 if(BUILD_TESTS)
   add_subdirectory(tests/ocltst)
Index: ROCm-OpenCL-Runtime-rocm-5.1.3/packaging/CMakeLists.txt
===================================================================
--- ROCm-OpenCL-Runtime-rocm-5.1.3.orig/packaging/CMakeLists.txt
+++ ROCm-OpenCL-Runtime-rocm-5.1.3/packaging/CMakeLists.txt
@@ -10,7 +10,6 @@
 set(CPACK_DEB_COMPONENT_INSTALL ON)
 set(CPACK_RPM_COMPONENT_INSTALL ON)

-install(TARGETS clinfo DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT binary)
 install(TARGETS amdocl DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT binary)
 install(FILES ${CMAKE_SOURCE_DIR}/LICENSE.txt DESTINATION ${CMAKE_INSTALL_DOCDIR} COMPONENT binary)
 
